src/interactivity/base/EventSynthesis.cpp

```diff
@@ -29,10 +29,22 @@ std::deque<std::unique_ptr<KeyEvent>> Microsoft::Console::Interactivity::CharToK
         // not include symbolic character for DBCS.
         WORD CharType = 0;
         GetStringTypeW(CT_CTYPE3, &wch, 1, &CharType);
+        bool c3Applicable = CharType != C3_NOTAPPLICABLE;
 
-        if (WI_IsFlagSet(CharType, C3_ALPHA) || GetQuickCharWidth(wch) == CodepointWidth::Wide)
```

Comment:
`GetQuickCharWidth` *never* returns `CodepointWidth::Wide` since #8035. Perhaps the correct fix is to restore the width detection logic?

Reply:
off: wonder if there's a static analysis rule for this. It seems quite obvious and pretty detectable.

Reply:
Also cc @DHowett because he created #8035

Reply:
It's really funny that you found this all at the same time. I didn't realize i was doing a parallel investigation in #9052. I mentioned you in it as well.

Reply:
I didn't realize what #9052 is about until this PR. This turns out to be a more general issue than I thought. Vim is not the only victim. Hope my investigation here helps a bit.